src/idc/idc.c

```diff
@@ -351,7 +351,6 @@ int idc_init(void)
 	tr_info(&idc_tr, "idc_init()");
 
 	/* initialize idc data */
-	*idc = rzalloc(SOF_MEM_ZONE_SYS, 0, SOF_MEM_CAPS_RAM, sizeof(**idc));
```

Comment:
How about non-Zephyr builds? Would this impact those as well?

Reply:
I suppose that non-Zephyr builds are not using definitions from zephyr/wrapper.c so they might be affected then.
I have only tested zephyr build. Do we have any CI test here that would check IDC for the non-zephyr configuration?
@kv2019i Shall I put an #ifndef ZEPHYR in such case or do you have any other suggestion on how to handle this?

Reply:
yes, I'd say - just move that allocation 5 lines up under the `#ifndef` there

Reply:
done